api.go

```diff
@@ -987,3 +988,228 @@ func (app *App) APIGetChallengeSkin() func(c echo.Context) error {
 		})
 	})
 }
+
+type APITokenResponse struct {
+	Token string `json:"token" example:"dfghjdrtjdrfr"`
+}
+
+type APILoginRequest struct {
+	Username string `json:"username" example:"notch"`
+	Password string `json:"password" example:"OIXtjQeASfz"`
+}
+
+// APILogin godoc
+//
+//	@Summary	  	Get a token
+//	@Description	Get a token for login credentials.
+//	@Tags		  		users, auth
+//	@Accept				json
+//	@Produce			json
+//	@Success			200	{object}	APITokenResponse
+//	@Failure			400	{object}	APIError
+//	@Failure			401	{object}	APIError
+//	@Failure			423	{object}	APIError
+//	@Failure			500	{object}	APIError
+//	@Router				/drasl/api/v2/login [post]
+func (app *App) APILogin() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		var req APILoginRequest
+		err := c.Bind(&req)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed JSON request")
+		}
+
+		var user User
```

Comment:
We should unify this logic with `FrontLogin` by moving it to a function in `user.go`, like how `CreateUser` and `DeleteUser` work. Could be done in a follow-up PR.